fileglancer/filestore.py

```diff
@@ -158,9 +159,13 @@ def yield_file_infos(self, path: Optional[str] = None) -> Generator[FileInfo, No
                 try:
                     yield self._get_file_info_from_path(entry_path)
                 except (FileNotFoundError, PermissionError):
+                    log.error(f"Error accessing entry: {entry_path}")
                     continue
         except (FileNotFoundError, PermissionError):
-            return
+            if PermissionError:
+                raise PermissionError(f"Permission denied for path: {full_path}")
+            elif FileNotFoundError:
```

Comment:
Looking at this again, I don't think it will work. The `if` isn't testing anything here, it's always true because PermissionError is truthy. The correct way to write this would be to have two `except` statements, one for each type. But I don't think the try/except is necessary at all, because the code calling this is only looking to catch these types, so if we remove the try/except they will be thrown. I will fix this in another PR I'm working on, but please let me know if I missed something.